java/README.md

```diff
@@ -0,0 +1,61 @@
+   - Standard Fields
+       - Label: `CarMaintenance`
+       - Plural Label: `CarMaintenances`
+   - Custom Fields
+       - `Cost` (Number)
+       - `Mileage` (Number)
+       - `WorkDescription` (Text, 200)
+7. For the examples in the `accountupdateapp` package, another custom **_NewAccount_** [platform event](https://developer.salesforce.com/docs/atlas.en-us.platform_events.meta/platform_events/platform_events_define_ui.htm) has to be created in the Salesforce org. [More info here](src/main/java/accountupdateapp/README.md).
+
+### Execution
+1. Update the configurations in the `src/main/resources/arguments.yaml` file. The required configurations will apply to all the examples and the optional ones depends on which example is being executed. The configurations include:
+   1. Required configurations:
+       * `PUBSUB_HOST`: Specify the Pub/Sub API endpoint to be used.
+       * `PUBSUB_PORT`: Specify the Pub/Sub API port to be used (usually 7443).
+       * `LOGIN_URL`: Specify the login url of the Salesforce org being used to run the examples.
+       * `USERNAME` & `PASSWORD`: For authentication via username and password, you will need to specify the username and password of the Salesforce org. 
+       * `ACCESS_TOKEN` & `TENANT_ID`: For authentication via session token an tenant ID, you will need to specify the sessionToken and tenant ID of the Salesforce org.
+   2. Optional Parameters:
+       * `TOPIC`: Specify the topic for which you wish to publish/subscribe. 
+       * `NUMBER_OF_EVENTS_TO_PUBLISH`: Specify the number of events to publish while using the PublishStream RPC.
+       * `NUMBER_OF_EVENTS_TO_SUBSCRIBE`: Specify the number of events to subscribe while using the SubscribeStream RPC.
+       * `USE_PLAINTEXT_CHANNEL`: Specify whether a plaintext channel has to be used.
+       * `USE_PROVIDED_LOGIN_URL`: The LOGIN_URL required parameter is usually translated into a specific my domain URL for connecting to the Salesforce org. In case the supplied LOGIN_URL has to be used, set this configuration to `true`.
+       * `REPLAY_PRESET`: Specify the ReplayPreset for subscribe examples.
+2. After setting up the configurations, any example can be executed using the `./run.sh` file available at the parent directory.
+   * Format for running the examples: `./run.sh <package_name>.<example_class_name>`
+   * Example: `./run.sh genericpubsub.PublishStream`
+
+## Implementation
+This repo can be used as a reference point for clients looking to create a Java app to integrate with Pub/Sub API. Note that the project structure and the examples included in this repo are intended for demo purposes only and clients are free to implement their own Java apps in any way they see fit. 
+
+# Limitations
+1. No support for C2C type of authentication. This may be included in future updates.
+2. No support for auth token refreshes - At some point the authentication token will expire. At this time, these examples do not handle re-authentication.
```

Comment:
Session tokens do not need auth refresh. Clients can retrieve a new session token if the existing session token is revoked by admin or through the connected app.